### DISCUSSION TITLE
Conditional relationship column

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#7235] [Help]
As mentioned in other questions, I am making an app for a friend who is a driving instructor.

I have a section where he can add lesson dates and select which pupil is having the lesson and similarly for test dates.

Once a pupil has passed their test, the database will be updated as such.

Is there a way to alter the relationship dropdown menu to emit the names of students who have passed their test? Since these students will no longer be required. 
However, We want to keep the details there so that we can measure pass rate etc.

**Comment by melohagan** [comment]
Hi @PaulMcF1987 

For the internal DB, unfortunately not. The relationship picker automatically pulls its dropdown values from the related table, and there is no filtering that can be applied to this process at the moment.
You may have more luck with an external DB as you can filter the relationships and run custom queries. 

For the internal DB, a potential but not so convenient work around could be to have and manage multiple tables:

One instructor -> Passing Pupils
One instructor -> Non-passing Pupils
One instructor -> All pupils

In this way you can have a relationship picker for the non-passing pupils.

**Reply by PaulMcF1987** [reply]
@melohagan Thank you for your reply.

All pupils would on the one table as all pupils start as not passed as they are currently under instruction.
The reason for wanting to filter the pupils is so that when the instructor is setting lessons for current pupils, he does not have t sift through pupils who have already passed and are no longer under his tuition.

2 pupils over the course of a few years could have the same name... if I could filter the pupils then it makes it less likely that he would come across this and accidentally book a lesson for a pupil who is no longer under his tuition.

**Reply by PaulMcF1987** [reply]
I have created [an enhancement request](https://github.com/Budibase/budibase/issues/7298)

Youll probably notice that a lot of my questions have been based around the relationship columns and I am generally getting the same sort of answers back.
I feel more freedom to use queries within the design pages (I.e being able to add them to option pickers etc) coould solve 99.9% of my issues, I cant speak for anyone else but it could assist others as well.

**Reply by melohagan** [reply]
The relationships are a bit awkward at times. Thanks for the feedback!

**Reply by PaulMcF1987** [reply]
@melohagan is a query type set up something that might be looked at going forward?
I was thinking... Where the option-picker options are entered, if there was the option to input a query instead that might work well. or even on the design page itself, once selecting the element, if there was an option to add a query

**Reply by melohagan** [reply]
You can use a Data Provider as the source of an options picker. The data provider could use a query as its source if that's what you mean?
